CRM/Contact/BAO/Query.php

```diff
@@ -5164,15 +5164,20 @@ public function dateQueryBuilder(
         $date = "'$date'";
       }
 
+      $this->_tables[$tableName] = $this->_whereTables[$tableName] = 1;
+
+      // handle IS NULL / IS NOT NULL / IS EMPTY / IS NOT EMPTY
+      if ($this->nameNullOrEmptyOp($fieldName, $op, $grouping)) {
```

Comment:
Yashodha, please ignore my earlier note. I read the code more carefully and I agree I can move that earlier as you suggest.